src/vumi2/applications/turn_channels_api/messages.py

```diff
@@ -66,27 +66,28 @@ def turn_outbound_from_msg(message: Message) -> dict:
     return {"messages": [{"id": message.message_id}]}
 
 
-def turn_event_from_ev(event: Event) -> dict:
+def turn_event_from_ev(event: Event, outbound: Message) -> dict:
     """
     From https://whatsapp.turn.io/docs/api/channel_api#sending-outbound-message-status-to-your-channel
 
     * id (str) - The UUID of the message the event is for.
     * timestamp (str) - The timestamp at which the event occurred.
     * status (dict) - The status of the event. Currently only sent, delivered, and read
       are supported.
+    * recipient_id (str) - The address the message was sent to.
     """
     ev = {
         "status": {
             "id": event.user_message_id,
             "timestamp": str(int(event.timestamp.timestamp())),
             "status": None,
+            "recipient_id": outbound.to_addr,
         }
     }
     if event.event_type == EventType.DELIVERY_REPORT:
```

Comment:
I think we need another check for `EventType.NACK`, since we only want `ack`s to give us a successful status, `nack`s should also be failed.